storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -304,87 +308,17 @@ public boolean containsBlock(final Bytes32 root) {
-  }
-
   public Optional<Bytes32> getBlockRootBySlot(final UnsignedLong slot) {
-    if (store == null || chainHead.isEmpty() || isHistoricalBlockRootAtSlotUnavailable(slot)) {
-      return Optional.empty();
-    }
-    return getBlockRootBySlot(slot, chainHead.get());
+    return chainHead.flatMap(head -> getBlockRootBySlot(slot, head.getRoot()));
   }
 
-  public Optional<Bytes32> getBlockRootBySlot(final UnsignedLong slot, final Bytes32 headRoot) {
-    if (store == null || isHistoricalBlockRootAtSlotUnavailable(slot)) {
-      return Optional.empty();
-    }
-    return store.getBlockAndState(headRoot).flatMap(head -> getBlockRootBySlot(slot, head));
-  }
-
-  private boolean isHistoricalBlockRootAtSlotUnavailable(final UnsignedLong slot) {
-    // If slot is prior to genesis, it is not available
-    final UnsignedLong genesisSlot = UnsignedLong.valueOf(Constants.GENESIS_SLOT);
-    if (slot.compareTo(genesisSlot) < 0) {
-      return true;
-    }
-
-    // If slot is out of range of the latest finalized block, it is unavailable
-    final UnsignedLong slotsPerHistoricalRoot = UnsignedLong.valueOf(SLOTS_PER_HISTORICAL_ROOT);
-    final UnsignedLong oldestAvailableState = store.getLatestFinalizedBlockSlot();
-    final UnsignedLong oldestAvailableRoot =
-        oldestAvailableState.compareTo(slotsPerHistoricalRoot) >= 0
-            ? oldestAvailableState.minus(slotsPerHistoricalRoot)
-            : UnsignedLong.ZERO;
-    return slot.compareTo(oldestAvailableRoot) < 0;
-  }
-
-  private Optional<Bytes32> getBlockRootBySlot(
-      final UnsignedLong slot, final SignedBlockAndState headBlock) {
-    checkNotNull(slot);
-    final UnsignedLong slotsPerHistoricalRoot = UnsignedLong.valueOf(SLOTS_PER_HISTORICAL_ROOT);
-    // Since older blocks are pruned - query the newest queryable slot where possible
-    final UnsignedLong youngestQueryableSlot = slot.plus(slotsPerHistoricalRoot);
-
-    SignedBlockAndState currentBlock = headBlock;
-    while (currentBlock != null) {
-      if (currentBlock.getSlot().compareTo(slot) <= 0) {
-        // The current block is less than or equal to the target slot, so it must be the block in
-        // effect
-        return Optional.of(currentBlock.getRoot());
-      }
-      if (BeaconStateUtil.isBlockRootAvailableFromState(currentBlock.getState(), slot)) {
-        // The root is available from the current state
-        return Optional.of(BeaconStateUtil.get_block_root_at_slot(currentBlock.getState(), slot));
-      }
-
-      // Pull an older block to search
-      final UnsignedLong oldestQueryableSlot = currentBlock.getSlot().minus(slotsPerHistoricalRoot);
-      final UnsignedLong olderSlot = max(youngestQueryableSlot, oldestQueryableSlot);
-      final Bytes32 olderRoot = get_block_root_at_slot(currentBlock.getState(), olderSlot);
-      currentBlock = store.getBlockAndState(olderRoot).orElse(null);
-    }
-
-    return Optional.empty();
+  public Optional<Bytes32> getBlockRootBySlot(
```

Comment:
Are we okay with not being able to get any block root before finalization?

Reply:
If we do want to use ProtoArray for collecting block roots pre-finalization, we could modify it in such a way that `findHead` (and thus `computeDeltas`) would not traverse through ProtoNodes pre-finalization, and stop at the finalization slot, but we'd still have access to information on nodes pre-finalization. This way ForkChoice could stay efficient.

Reply:
There was always a limit to how far back we could go with `RecentChainData` anyway so I think it's fine if it cuts off at the finalised block.  We don't have the blocks or states before that anyway.